Functor.md

```diff
@@ -0,0 +1,150 @@
+# Intro
+
+Welcome to this series of videos about Functional Programming in Kotlin with Arrow. In this video, we will talk about the
+Functor Typeclass. Whenever you find yourself in a situation where you need to map over or transform your data, you will want
+to use the Functor.
+
+# Slide 1
+Functor is available in the arrow-typeclasses module.
+```
+// gradle
+compile 'io.arrow-kt:arrow-typeclasses:$arrowVersion'
+
+import arrow.typeclasses.Functor
+
+```
+
+# Slide 2
+`Functor` is a **Typeclass**, so it defines a given behavior.
+`Applicative` and `Monad` inherit its combninators. You will learn more about those
+in further videos.
+
+Functor
+Applicative
+Monad
+
+# Slide 3
+```
+interface Functor<F> {
+    //...
+}
+```
+* The `F` on the declaration stands for a **type constructor**. That's why `Functor`is considered **polymorphic**.
+
+# Slide 4
+`Functor` abstracts the hability to **map** over the computational context of a type constructor `F`.
+In other words, it provides a mapping function for the `F` type:
+```
+fun F<A>.map(f: (A) -> B): F<B>
+```
+
+# Slide 5
+```
+fun F<A>.map(f: (A) -> B): F<B>
+```
+* `f` is the function transforming the wrapped value.
+* `map()` returns the transformed value wrapped into the same context: `F<A> -> F<B>`
+
+# Slide 6
+Any type constructor whose contents can be transformed can provide an instance of `Functor`.
+* `Option<A>`
+* `Try<A>`
+* `List<A>`
+* `IO<A>`
+* ...and many more.
+
+# Slide 7
+You can request the functor instance statically from any type constructor supporting it:
```

Comment:
The alternative to request the function statically and the most important perhaps to showcase in this slides is the ability to write polymorphic programs that are abstract and then made concrete:
```kotlin
fun <F> Functor<F>.addOne(fa: Kind<F, Int>): Kind<F, Int> =
  fa.map { it + 1 }

Option.functor().addOne(Option(1)).fix() // Option(2)
Try.functor().addOne(Try { 1 }).fix() // Success(2)
```